lineapy/graph_reader/node_collection.py

```diff
@@ -0,0 +1,240 @@
+                    if nid in input_parameters_node.keys()
+                ]
+            )
+        )
+
+    def _update_graph(self, graph: Graph) -> None:
+        """
+        Update graph_segment class member based on node_list
+
+        Need to manually run this function at least once if you need the graph object
+        for code generation
+        """
+        self.graph_segment = graph.get_subgraph_from_id(list(self.node_list))
+        self.raw_codeblock = get_source_code_from_graph(
+            self.graph_segment
+        ).__str__()
+        self.is_empty = self.raw_codeblock == ""
+
+    def get_function_definition(self, indentation=4) -> str:
+        """
+        Return a codeblock to define the function of the graph segment
+        """
+
+        indentation_block = " " * indentation
+        artifact_codeblock = "\n".join(
+            [
+                f"{indentation_block}{line}"
+                for line in self.raw_codeblock.split("\n")
+                if len(line.strip(" ")) > 0
+            ]
+        )
+        name = self.safename
+        args_string = ", ".join(sorted([v for v in self.input_variables]))
+        return_string = ", ".join([v for v in self.return_variables])
+        return f"def get_{name}({args_string}):\n{artifact_codeblock}\n{indentation_block}return {return_string}"
+
+    def get_function_call_block(
+        self, indentation=0, keep_lineapy_save=False, result_placeholder=None
+    ) -> str:
+        """
+        Return a codeblock to call the function with return variables of the graph segment
+
+        :param int indentation: indentation size
+        :param bool keep_lineapy_save: whether do lineapy.save() after execution
+        :param Optional[str] result_placeholder: if not null, append the return result to the result_placeholder
+
+        The result_placeholder is a list to capture the artifact variables right
+        after calculation. Considering following code,
+
+        a = 1
+        lineapy.save(a,'a')
+        a+=1
+        b = a+1
+        lineapy.save(b,'b')
+        c = a+1
+        lineapy.save(c,'c')
+
+        we need to record the artifact a before it is mutated.
+
+        """
+
+        indentation_block = " " * indentation
+        return_string = ", ".join(self.return_variables)
+        args_string = ", ".join(sorted([v for v in self.input_variables]))
+
+        codeblock = f"{indentation_block}{return_string} = get_{self.safename}({args_string})"
+        if (
+            keep_lineapy_save
+            and self.collection_type == NodeCollectionType.ARTIFACT
+        ):
+            codeblock += f"""\n{indentation_block}lineapy.save({self.return_variables[0]}, "{self.name}")"""
+        if result_placeholder is not None:
+            codeblock += f"""\n{indentation_block}{result_placeholder}.append(copy.deepcopy({self.return_variables[0]}))"""
+
+        return codeblock
+
+    def get_import_block(self, indentation=0) -> str:
```

Comment:
Fix the existing get_import_block function and use it here instead of redefining it.